functions.php

```diff
@@ -280,4 +279,67 @@ function validateFormSignUp(mysqli $link, array $registration): array {
     return $errors;
 }
 
+function getUserDb(mysqli $link, string $email): array {
+	$sql = "SELECT * FROM user WHERE email = ?";
+    $stmt = mysqli_prepare($link, $sql);
+    mysqli_stmt_bind_param($stmt, 's', $email);
+    mysqli_stmt_execute($stmt);
+    $result = mysqli_stmt_get_result($stmt);
+    $user = mysqli_fetch_array($result, MYSQLI_ASSOC);
+
+    return $result && !empty($user) ? $user : [];
+}
+
+function validateFormLogin(mysqli $link, array $login, mixed $user): array {
+    $requiredFields = ['email', 'password'];
+
+    $errors = [];
+    foreach ($login as $key => $value) {
+        if (in_array($key, $requiredFields) && empty($value)) {
+            $errors[$key] = 'Поле надо заполнить';
+        }
+    }
+
+    if (!$user && empty($errors['email'])) {
+		    $errors['email'] = 'Такой пользователь не найден';
+	}
+
+    return $errors;
+}
+
+function getSessionName(): string {
+    return $_SESSION['user']['name'] ?? '';
+}
+
+function errorPage(array $sqlCategories, string $userName): void{
+    $pageContent = include_template('403.php', ['sqlCategories' => $sqlCategories,]);
+
+    $layoutContent = include_template('layout.php', [
+        'content' => $pageContent,
+        'categories' => $sqlCategories,
+        'title' => 'Доступ запрещен',
+        'userName' => $userName,
+    ]);
+
+    print($layoutContent);
+    exit();
+}
+
+function notFoundPage(array $sqlCategories, string $userName): void{
+    $pageContent = include_template('404.php', ['sqlCategories' => $sqlCategories,]);
+
+    $layoutContent = include_template('layout.php', [
+        'content' => $pageContent,
+        'categories' => $sqlCategories,
+        'title' => 'Страница не найдена',
+        'userName' => $userName,
+    ]);
+
+    print($layoutContent);
+    exit();
+}
+
+function checkPassword(array $login, array $user): bool {
+    return password_verify($login['password'], $user['password']) ? true : false;
```

Comment:
Функция password_verify уже возвращает булево
```suggestion
    return password_verify($login['password'], $user['password']);
```